File: algorithms/591. Tag Validator/solution.cc
class Solution {
public:
    bool isValid(string code) {
        // inspired by stefanpochmann
        regex cdata("<!\\[CDATA\\[.*?\\]\\]>"), tag("<([A-Z]{1,9})>[^<]*</\\1>");
        code = regex_replace(code, cdata, "c");
        string pre = code;
        while ( pre != (code = regex_replace(code, tag, "t")) ) pre = code;
        return "t" == code;
    }
};

File: algorithms/591. Tag Validator/solution.py
class Solution(object):
    def isValid(self, code):
        """
        :type code: str
        :rtype: bool
        """
        # @stefanpochmann
        code = re.sub(r"<!\[CDATA\[.*?\]\]>", "c", code)
        pre = None
        while pre != code:
            pre = code
            code = re.sub(r"<([A-Z]{1,9})>[^<]*</\1>", "t", code)
        return code == 't'
